src/search.cpp

```diff
@@ -1214,7 +1215,8 @@ namespace {
           && !InCheck
           && !givesCheck
           &&  futilityBase > -VALUE_KNOWN_WIN
-          && !pos.advanced_pawn_push(move))
```

Comment:
Why remove these?  Inlining these conditions hurts readability here IMO.  advanced_pawn_push is a good name for what the function does.